File: Groupe1/TP2/src/data_structure.py
# Author:    Rémi Jara
# Created:   17/09/2024

class Student:
    def __init__(self):
        self.firstName = ""
        self.lastName = ""
        self.address = ""
        self.grades = [0.0] * 5

students = []

for i in range(5):
    student = Student()

    student.firstName = input(f"Enter the first name of student {i + 1}: ")
    student.lastName = input(f"Enter the last name of student {i + 1}: ")
    student.address = input(f"Enter the address of student {i + 1}: ")

    for j in range(5):
        student.grades[j] = float(input(f"Enter the grade {j + 1} of student {i + 1}: "))

    students.append(student)

for i, student in enumerate(students):
    print(f"Student {i + 1}")
    print(f"First name: {student.firstName}")
    print(f"Last name: {student.lastName}")
    print(f"Address: {student.address}")
    print(f"Grades: {' '.join(str(grade) for grade in student.grades)}")

File: Groupe1/TP2/src/data_structure.c
/**
 * Author:    Rémi Jara
 * Created:   17/09/2024
 *
 **/

struct Student
{
    char firstName[50];
    char lastName[50];
    char address[100];
    float grades[5];
};
#include <stdio.h>
#include <string.h>

int main()
{
    struct Student students[5];

    char firstName[50];
    char lastName[50];
    char address[100];
    float grades[5];

    for (int i = 0; i < 5; i++)
    {
        printf("Enter the first name of student %d: ", i + 1);
        scanf("%s", firstName);
        strcpy(students[i].firstName, firstName);

        printf("Enter the last name of student %d: ", i + 1);
        scanf("%s", lastName);
        strcpy(students[i].lastName, lastName);

        printf("Enter the address of student %d: ", i + 1);
        scanf("%s", address);
        strcpy(students[i].address, address);

        for (int j = 0; j < 5; j++)
        {
            printf("Enter the grade %d of student %d: ", j + 1, i + 1);
            scanf("%f", &grades[j]);
            students[i].grades[j] = grades[j];
        }
    }

    for (int i = 0; i < 5; i++)
    {
        printf("Student %d\n", i + 1);
        printf("First name: %s\n", students[i].firstName);
        printf("Last name: %s\n", students[i].lastName);
        printf("Address: %s\n", students[i].address);
        printf("Grades: ");
        for (int j = 0; j < 5; j++)
        {
            printf("%.2f ", students[i].grades[j]);
        }
        printf("\n");
    }

    return 0;
}